src/EditorFeatures/CSharpTest/Formatting/CodeCleanupTests.cs

```diff
@@ -568,6 +573,189 @@ public void VerifyAllVisualBasicCodeStyleFixersAreSupportedByCodeCleanup(string
+        {
+            case 1:
+            case 10:
+                break;
+            default:
+                break;
+        }
+
+        switch (y)
+        {
+            case 1:
+                break;
+            case 1000:
+            default:
+                break;
+        }
+
+        switch (x)
+        {
+            case 1:
+                break;
+            case 1000:
+                break;
+        }
+
+        switch (y)
+        {
+            default:
+                break;
+        }
+
+        switch (y) { }
+
+        switch (x)
+        {
+            case :
+            case 1000:
+                break;
+        }
+    }
+}
+";
+
+        private const string _fixed = @"
+class C
+{
+    public void M1(int x, int y)
+    {
+        switch (x)
+        {
+            case 1:
+            case 10:
+                break;
+        }
+
+        switch (y)
+        {
+            case 1:
+                break;
+        }
+
+        switch (x)
+        {
+            case 1:
+                break;
+            case 1000:
+                break;
+        }
+
+        switch (y)
+        {
+        }
+
+        switch (y) { }
+
+        switch (x)
+        {
+            case :
+            case 1000:
+                break;
+        }
+    }
+}
+";
+
+        [Fact]
+        [Trait(Traits.Feature, Traits.Features.CodeCleanup)]
+        public async Task RunThirdPartyFixer()
+        {
+            await TestThirdPartyCodeFixer<TestThirdPartyCodeFixWithFixAll, CaseTestAnalyzer>(_code, _fixed);
+        }
+
+        [Fact]
+        [Trait(Traits.Feature, Traits.Features.CodeCleanup)]
+        public async Task DoNotRunThirdPartyFixerWithNoFixAll()
+        {
+            await TestThirdPartyCodeFixer<TestThirdPartyCodeFixWithOutFixAll, CaseTestAnalyzer>(_code, _code);
+        }
+
+        [Theory]
+        [InlineData(DiagnosticSeverity.Warning)]
+        [InlineData(DiagnosticSeverity.Error)]
+        [Trait(Traits.Feature, Traits.Features.CodeCleanup)]
+        public async Task RunThirdPartyFixerWithSeverityOfWarningOrHigher(DiagnosticSeverity severity)
+        {
+            await TestThirdPartyCodeFixer<TestThirdPartyCodeFixWithFixAll, CaseTestAnalyzer>(_code, _fixed, severity);
+        }
+
+        [Theory]
+        [InlineData(DiagnosticSeverity.Hidden)]
+        [InlineData(DiagnosticSeverity.Info)]
+        [Trait(Traits.Feature, Traits.Features.CodeCleanup)]
+        public async Task DoNotRunThirdPartyFixerWithSeverityLessThanWarning(DiagnosticSeverity severity)
+        {
+            await TestThirdPartyCodeFixer<TestThirdPartyCodeFixWithOutFixAll, CaseTestAnalyzer>(_code, _code, severity);
+        }
+
+        [Fact]
+        [Trait(Traits.Feature, Traits.Features.CodeCleanup)]
+        public async Task DoNotRunThirdPartyFixerIfItDoesNotSupportDocumentScope()
+        {
+            await TestThirdPartyCodeFixer<TestThirdPartyCodeFixDoesNotSupportDocumentScope, CaseTestAnalyzer>(_code, _code);
```

Comment:
Might be nice to have a separate functions for when the code is expected to change and not change, instead of passing the same parameter for each.

`TestThirdPartyCodeFixerApplied<>` & `TestThirdPartyCodeFixerNoChanges<>` perhaps?